pyteal/ast/opup.py

```diff
@@ -29,6 +30,15 @@ class OpUpMode(Enum):
     OnCall = 1
 
 
+class OpUpFeePayer(Enum):
+    # Only the excess fee on the outer group should be used (set inner_tx.fee=0)
+    Excess = 0
+    # The app account will cover fees
+    Self = 1
```

Comment:
@barnjamin A couple of thoughts here.  Treat optionally - If you prefer as is, I can work with the PR as shown.  I can also speak live if helpful.
* Since the enum describes a payer, I'd opt for names that reference each actor like Caller (0), App (1), and PreferCaller (2). 
* The inner transaction fee payer concern exists beyond OpUp.  Perhaps the enum can be defined more generally as an enrichment to `InnerTxnBuilder`.

Reply:
Maybe I should rename to `fee_source` since the outer group excess isn't exactly an actor. Open to name change but I think we should treat it as where to pull fees from rather than trying to name actors. wdyt?

I don't immediately see how the enum might be used in an InnerTxn.  It's useful here because the inner transactions are constructed outside the authors immediate control but during construction of an InnerTxn presumably the author has much more control of how they set the fees?

Reply:
I've made changes to the name to avoid `payer` in favor of `source`, happy to chat about this

Reply:
@barnjamin Good to resolve once the _group excess_ naming question (below) is responded to.

>  I think we should treat it as where to pull fees from 

* Seems reasonable to me.  Thanks for additional framing context.
* Is _group excess_ a term familiar to most builders?  
  * If so, then let's keep as is.  
  * I don't necessarily have a _good_ alternative.  go-algorand uses the term _fee credit_.  The docs also discuss _fee pooling_.  If one of these terms seems better, then let's take it, otherwise let's keep as is.

> don't immediately see how the enum might be used in an InnerTxn. 

* I'm good to keep as is (specific to OpUp) though I'd like to do a quick verbal to socialize the thought.
* From my perspective, configuring the inner transaction payer applies to _any_ inner transaction construction.  We can discuss it out-of-band to the PR to see if it's applicable + worth pursuing.

Reply:
If we use `credit` elsewhere ill tweak to use the same term, no reason to come up with new ones

Reply:
@barnjamin Resolving - _credit_ works for me.  We can pick up an enriched inner transaction builder another time.